client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -195,8 +199,13 @@ const Checkout = React.createClass( {
 			return selectedSiteSlug
 				? `/plans/${ selectedSiteSlug }/thank-you`
 				: '/checkout/thank-you/plans';
-		} else if ( cart.create_new_blog && cartItems.hasDomainRegistration( cart ) && ! cartItems.hasPlan( cart ) ) {
-			const domainName = cartItems.getDomainRegistrations( cart )[ 0 ].meta;
+		} else if ( cart.create_new_blog && ! cartItems.hasPlan( cart ) ) {
+			let domainName;
+			if ( cartItems.hasDomainRegistration( cart ) ) {
+				domainName = cartItems.getDomainRegistrations( cart )[ 0 ].meta;
+			} else {
+				domainName = cartItems.getDomainRegistrations( this.state.previousCart )[ 0 ].meta;
```

Comment:
The list of purchases from this receipt is stored in `this.props.transaction.step.data.receipt.purchases` once the transaction response comes in. What do you think about pulling the domain `meta` out of there instead of checking between two states of the cart?